Assets/_Functionalities/ObjImporter/Scripts/ObjSpawner.cs

```diff
@@ -43,28 +44,44 @@ public void LoadProperties(List<LayerPropertyData> properties)
             this.propertyData = propertyData;
         }
 
-        private void StartImport()
+        private void Start()
         {
+            StartImport(); //called after loading properties or after setting the file path through the import adapter
+        }
+
+        public void StartImport()
+        {
+            if(importedObject)
+                Destroy(importedObject);
+            
             DisposeImporter();
 
             importer = Instantiate(importerPrefab);
 
-            var localPath = propertyData.ObjFile.LocalPath.TrimStart('/', '\\');
-            var path = Path.Combine(Application.persistentDataPath, localPath);
-
-            ImportObj(path);
+            var objPath = GetObjPathFromPropertyData();
+            var mtlPath = GetMtlPathFromPropertyData();
+            
+            ImportObj(objPath, mtlPath);
         }
 
-        private void ImportObj(string path)
+        private void ImportObj(string objPath, string mtlPath = "")
         {
             // the obj-importer deletes the obj-file after importing.
             // because we want to keep the file, we let the importer read a copy of the file
             // the copying can be removed after the code for the importer is changed
-            string copiedFilename = path + ".temp";
-            File.Copy(path, copiedFilename);
+            string copiedObjFilename = objPath + ".temp";
```

Comment:
is de objPath hier altijd != null?

Reply:
ja, anders moet dit script niet bestaan, of bedoel je dat niet?

Reply:
nee als om wat voor reden bijvoorbeeld de propertydata uri corrupt is of mist, dan zou het hier verkeerd gaan, misschien een logwarning hier